src/rebar_prv_dialyzer.erl

```diff
@@ -382,6 +384,29 @@ run_dialyzer(State, Opts) ->
             {0, State}
     end.
 
+filter_warnings([], Warnings) ->
+    Warnings;
+filter_warnings(Ignores, Warnings) ->
+    lists:filter(fun(Warning) ->
+        not lists:any(fun (Ignore) -> match_warning(Ignore, Warning) end, Ignores)
+    end, Warnings).
+
+match_warning({MatchTag, {MatchPath, MatchLine}, MatchMsg}, {Tag, {Path, Line}, Msg}) when MatchPath =/= '_' ->
+    lists:suffix(MatchPath, Path) andalso match_term([MatchTag, MatchLine, MatchMsg], [Tag, Line, Msg]);
```

Comment:
Dialyzer warnings include the absolute path in rebar3 because rebar3 compiles files using the absolute path. Dialyzer itself only actually uses whatever path the beam file has listed as its source file. This means it might just be "module.erl", a relative path (e.g. "src/module.erl") or the absolute path. For example a dialyzer error in a beam file from OTP itself would be use the base name "module.erl". If rebar3 changes how it supplies the source file to `compile:file/2` the path might change.

I was suggesting transforming the warning, before matching, something like:
```erlang
{Tag, {Path, Line}, Info} = Warning,
{Tag, {filename:basename(Path), Line}, Info}.
```

Please can we discuss this more in https://github.com/rebar/rebar/pull/460. It would not be ideal to concurrently introduce the same configuration option and have it work differently.